spec/jobs/modsulator_job_spec.rb

```diff
@@ -106,12 +106,81 @@
     end
   end
 
+  describe 'accessioned' do
+    (0..9).each do |i|
+      it "returns true for DOR objects that are acccessioned, false otherwise (:status_code #{i})" do
+        m = double
+        allow(m).to receive(:status_info).and_return({ :status_code => i })
+        if i == 6 || i == 7 || i == 8
+          expect(@mj.accessioned(m)).to be_truthy
+        else
+          expect(@mj.accessioned(m)).to be_falsy
+        end
+      end
+    end
+  end
+
   describe 'generate_xml_filename' do
     it 'creates a new filename using the correct convention' do
       expect(@mj.generate_xml_filename('/tmp/generate_xml_filename.xml')).to eq('generate_xml_filename-' + Settings.BULK_METADATA.XML + '.xml')
     end
   end
 
+  describe 'commit_new_version' do
+    let(:dor_test_object) { double('dor_item')}
+
+    it 'opens a new minor version with filename and username' do
+      expect(dor_test_object).to receive(:open_new_version).with({
+        vers_md_upd_info: {
+          significance: 'minor',
+          description: 'Descriptive metadata upload from testfile.xlsx',
+          opening_user_name: 'username'
+        }
+      }).and_return(true)
+      @mj.commit_new_version(dor_test_object, 'testfile.xlsx', 'username')
+    end
+  end
+
+  describe 'version_object' do
+    before :each do
+      @dor_object = double(pid: 'druid:123abc')
+      @workflow = double('workflow')
+      @log = double('log')
+    end
+
+    it 'writes a log error message if a version cannot be opened' do
+      expect(@dor_object).to receive(:status_info).at_least(:once).and_return({status_code: 6})
+      expect(DorObjectWorkflowStatus).to receive(:new).with(@dor_object.pid).and_return(@workflow)
+      expect(@workflow).to receive(:can_open_version?).and_return(false)
+      expect(@log).to receive(:puts).with("argo.bulk_metadata.bulk_log_unable_to_version #{@dor_object.pid}")
+
+      @mj.version_object(@dor_object, 'any_filename', 'any_user', @log)
+    end
+
+    it 'does not update the version if the object is in the registered state' do
+      expect(@dor_object).to receive(:status_info).at_least(:once).and_return({status_code: 1})
+      expect(@mj).not_to receive(:commit_new_version)
+
+      @mj.version_object(@dor_object, 'any_filename', 'any_user', @log)
+    end
+
+    it 'does not update the version if the object is in the opened state' do
+      expect(@dor_object).to receive(:status_info).at_least(:once).and_return({status_code: 1})
```

Comment:
status_code 1 is indeed both opened and registered.  i think the idea is that registered is a special case of opened.  i agree that the way it's implemented is confusing, but IIRC, there is logic that currently depends on the two being the same, and it would take a bit of effort to understand how to untangle things.  i think i tried giving opened and registered separate status codes at one point, and the tests started failing, and i declined to investigate further at the time.